### PR TITLE
fix sensitive data in logs found by GH code scanning

### PR DESCRIPTION
**Pull request by szuecs**
Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: ratelimit/ratelimit.go [hunk 1]
```diff
@@ -431,3 +432,7 @@ func Headers(s *Settings, retryAfter int) http.Header {
 		RetryAfterHeader: []string{strconv.Itoa(retryAfter)},
 	}
 }
+
+func getHashedKey(clearText string) string {
+	return fmt.Sprintf("%x", sha256.Sum256([]byte(clearText)))
```
Comment (AlexanderYastrebov) [inline ratelimit/ratelimit.go:437]: lets use a chance and change to https://golang.org/pkg/encoding/hex/#EncodeToString

Comment (szuecs) [reply ratelimit/ratelimit.go:437]: It’s not easily possible because we get an array not a slice so Sprintf is the simpler one.

Comment (AlexanderYastrebov) [reply ratelimit/ratelimit.go:437]:
```go
h := sha256.Sum256([]byte(clearText))
return hex.EncodeToString(h[:])
```

Comment (szuecs) [reply ratelimit/ratelimit.go]: The change showed in a benchmark `3->2` allocs and speedup of 2x for this little function.

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**